File: model/src/main/scala/ru/mipt/acsl/decode/modeling/impl/Message.scala
package ru.mipt.acsl.decode.modeling.impl

import ru.mipt.acsl.decode.modeling._
import ru.mipt.acsl.decode.modeling.aliases.TransformationMessage

/**
  * @author Artem Shein
  */
case class Message(level: Level, text: String) extends TransformationMessage {

  def error(msg: String): TransformationMessage = message(ErrorLevel, msg)

  def warning(msg: String): TransformationMessage = message(Warning, msg)

  def notice(msg: String): TransformationMessage = message(Notice, msg)

  def message(level: Level, msg: String): TransformationMessage = Message(level, msg)
}

File: model/src/main/java/ru/mipt/acsl/decode/model/Message.java
package ru.mipt.acsl.decode.model;

import ru.mipt.acsl.decode.model.proxy.Level;

/**
 * @author Artem Shein
 */
public interface Message {

    static Message newError(String text) {
        return new MessageImpl(Level.ERROR, text);
    }

    static Message newInstance(Level level, String text) {
        return new MessageImpl(level, text);
    }

    Level level();

    String text();

}
